train_nsf_sim_cache_sid_load_pretrain.py

```diff
@@ -62,27 +58,16 @@ def record(self):
         return f"[{current_time}] | ({elapsed_time_str})"
 
 
-def main():
-    n_gpus = torch.cuda.device_count()
-    if torch.cuda.is_available() == False and torch.backends.mps.is_available() == True:
-        n_gpus = 1
-    os.environ["MASTER_ADDR"] = "localhost"
-    os.environ["MASTER_PORT"] = str(randint(20000, 55555))
-    children = []
-    for i in range(n_gpus):
-        subproc = mp.Process(
-            target=run,
-            args=(
-                i,
-                n_gpus,
-                hps,
-            ),
-        )
-        children.append(subproc)
-        subproc.start()
+def train_nsf_etc(arg_string):
+    hps = utils.get_hparams(args_list=arg_string.split(" "))
+    assert hps.version == "v1" # Import is not correct if this is not the case.
+    os.environ["CUDA_VISIBLE_DEVICES"] = hps.gpus.replace("-", ",") # TODO: remove?
+    n_gpus = len(hps.gpus.split("-"))
 
-    for i in range(n_gpus):
-        children[i].join()
+    assert torch.cuda.device_count() == 1
+    os.environ["MASTER_ADDR"] = "localhost" # TODO: remove?
```

Comment:
this is related to the gradio they set up I think
probably can be removed

Reply:
It's actually required by `torch.distributed` (also the PORT).

Reply:
but you don't need that anymore either right?

Reply:
I do. It fails without these env variables.